src/core/execute.h

```diff
@@ -316,6 +316,7 @@ struct ExecContext {
         ProtectHome protect_home;
         bool protect_hostname;
         bool mount_apivfs;
+	bool memory_ksm;
```

Comment:
so you expose this as a simple bool. and allow it to be enabled only with that. Shouldn't this be a tri-state? i.e. an int value, that is either < 0, == 0 or > 0 for indicate "don't touch", "disable", "enable". We use those at many places across the codebase.

Or to say this differently: wouldn't a scenario make sense wher this option is turned off for pid1 itself, but then shall be turned off for a specific service?

Reply:
The goal is to be able to enable/disable this feature for different services.

Reply:
make this an "int" hence

Reply:
and then initialize it to -1 to exec_context_init()